ffi.c

```diff
@@ -158,7 +158,7 @@ static int64_t check_intptr(lua_State* L, int idx, void* p, struct ctype* ct)
         ct->base_size = 8;
         ct->type = INT64_TYPE;
         ct->is_defined = 1;
-        ret = luaL_checknumber(L, idx);
+        ret = (int64_t)luaL_checknumber(L, idx);
```

Comment:
may be for Lua >= 5.3 use lua_Integer type if it has 64. Not sure what best way to do it with preprocessor only.